src/Tracy/Dumper.php

```diff
@@ -415,26 +415,31 @@ public static function encodeString($s, $maxLength = NULL)
 			$table["\t"] = '\t';
 		}
 
-		if (preg_match('#[^\x09\x0A\x0D\x20-\x7E\xA0-\x{10FFFF}]#u', $s) || preg_last_error()) {
+		if (function_exists('mb_substr')) {
+			if ($maxLength && strlen($s) > $maxLength) {
+				$s = mb_substr($tmp = $s, 0, $maxLength, 'UTF-8');
+				$shortened = $s !== $tmp;
+			} else {
+				$shortened = false;
+			}
```

Comment:
This can be simplified by initializing `$shortend` variable first to `FALSE` (note the uppercase letters)

~~~php
$shortened = FALSE;
if ($maxLength && strlen($s) > $maxLength && function_exists('mb_substr')) {
	$s = mb_substr($tmp = $s, 0, $maxLength, 'UTF-8');
	$shortened = $s !== $tmp;
}
~~~